documentation/htc_workloads/specific_resources/openmpi-jobs.md

```diff
@@ -12,54 +12,65 @@ this, as long as the core count is small (4 is known to work well, 8 and 16
 becomes more difficult due to the limited number of resources).
 
 
-To get started, first compile your code using the OpenMPI in the modules
-system. For example:
+To get started, first compile your code using the OpenMPI container. You can create your own OpenMPI container or use the one that is available on dockerhub. OSG has a `openmpi` container. To compile your code using the OSG provided image, load the imagae first. 
 
-    $ module load openmpi
+    $ apptainer shell /ospool/uc-shared/public/OSG-Staff/openmpi.sif
     $ mpicc -o hello hello.c 
 
-
-You can test the executable locally using `mpiexec`:
+The `hello.c` is an example hello world code that can be executed using multiple processors. The code is given below:
+
+```
+#include <mpi.h>
+#include <stdio.h>
+
+int main(int argc, char** argv) {
+        MPI_Init(NULL, NULL);
+        int world_size;
+        MPI_Comm_size(MPI_COMM_WORLD, &world_size);
+        int world_rank;
+        MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);
+        char processor_name[MPI_MAX_PROCESSOR_NAME];
+        int name_len;
+        MPI_Get_processor_name(processor_name, &name_len);
+        printf("Hello world from processor %s, rank %d out of %d processors\n", processor_name, world_rank, world_size);
+        MPI_Finalize();
+}
+```
+After compiling the code, you can test the executable locally using `mpiexec`:
 
     $ mpiexec -n 4 hello
-    Hello world from process 1 of 4
-    Hello world from process 3 of 4
-    Hello world from process 0 of 4
-    Hello world from process 2 of 4
-
+    Hello world from processor ap21.uc.osg-htc.org, rank 0 out of 4 processors
+    Hello world from processor ap21.uc.osg-htc.org, rank 1 out of 4 processors
+    Hello world from processor ap21.uc.osg-htc.org, rank 2 out of 4 processors
+    Hello world from processor ap21.uc.osg-htc.org, rank 3 out of 4 processors
 
```

Comment:
Add a comment about exiting from the container after you compile/test.

Reply:
Done